src/flags/layout.rs

```diff
@@ -0,0 +1,163 @@
+//! This module defines the [Layout] flag. To set it up from [ArgMatches], a [Yaml] and its
+//! [Default] value, use its [configure_from](Configurable::configure_from) method.
+
+use super::Configurable;
+
+use crate::config_file::Config;
+
+use clap::ArgMatches;
+use yaml_rust::Yaml;
+
+/// The flag showing which output layout to print.
+#[derive(Clone, Debug, Copy, PartialEq, Eq)]
+pub enum Layout {
+    Grid,
+    Tree,
+    OneLine,
+}
+
+impl Configurable<Layout> for Layout {
+    /// Get a potential `Layout` variant from [ArgMatches].
+    ///
+    /// If any of the "tree", "long" or "oneline" arguments is passed, this returns the
+    /// corresponding `Layout` variant in a [Some]. Otherwise if the number of passed "blocks"
+    /// arguments is greater than 1, this also returns the [OneLine](Layout::OneLine) variant.
+    /// Finally if neither of them is passed, this returns [None].
+    fn from_arg_matches(matches: &ArgMatches) -> Option<Self> {
+        if matches.is_present("tree") {
+            Some(Self::Tree)
+        } else if matches.is_present("long")
+            || matches.is_present("oneline")
+            || matches.is_present("inode")
+            || matches!(matches.values_of("blocks"), Some(values) if values.len() > 1)
+        // TODO: handle this differently
```

Comment:
Hey, anything you were planning to change?

Reply:
@meain maybe but that is more something that could use some redesigning on how the logic works in general. I wouldn't say that this is part of the scope of this merge request.